core/src/program/blocks/span_block.rs

```diff
@@ -38,21 +39,26 @@ const MAX_OPS_PER_BATCH: usize = GROUP_SIZE * BATCH_SIZE;
 pub struct Span {
     op_batches: Vec<OpBatch>,
     hash: Digest,
+    decorators: DecoratorList,
 }
 
 impl Span {
     // CONSTRUCTOR
     // --------------------------------------------------------------------------------------------
-    /// Returns a new [Span] block instantiated with the specified operations.
+    /// Returns a new [Span] block instantiated with the specified operations and decorators.
     ///
     /// # Errors (TODO)
     /// Returns an error if:
     /// - `operations` vector is empty.
     /// - `operations` vector contains any number of system operations.
-    pub fn new(operations: Vec<Operation>) -> Self {
+    pub fn new(operations: Vec<Operation>, decorators: DecoratorList) -> Self {
```

Comment:
Maybe it is better to add a new constructor - e.g., `with_decorators()`?

Reply:
Added.